sharktank_models/test_suite/benchmarks/test_model_benchmark.py

```diff
@@ -138,60 +110,79 @@ def setup_class(self, pytestconfig):
             
             # retrieving golden values
             self.golden_time_tolerance_multiplier = data.get("golden_time_tolerance_multiplier", {}).get(sku)
-            self.golden_time = data.get("golden_time", {}).get(sku)
+            self.golden_time = data.get("golden_time_ms", {}).get(sku)
             self.golden_dispatch = data.get("golden_dispatch", {}).get(sku)
             self.golden_size = data.get("golden_size", {}).get(sku)
             
             # Custom configurations
-            self.specific_rocm_chip_to_ignore = data.get("specific_rocm_chip_to_ignore", [])
+            self.specific_chip_to_ignore = data.get("specific_chip_to_ignore", [])
             self.real_weights_file_name = data.get("real_weights_file_name", "real_weights.irpa")
 
             # custom configurations related to e2e testing
             self.compilation_required = data.get("compilation_required", False)
             self.compiled_file_name = data.get("compiled_file_name")
-            self.compile_flags = data.get("compile_flags", [])
             self.mlir_file_path = data.get("mlir_file_path", "")
             self.modules = data.get("modules", [])
-
-
-    def test_rocm_benchmark(self):
+            
+            # ROCM or CPU specific configurations
+            self.compile_flags = data.get("compile_flags", [])
+            self.benchmark_additional_flags = []
+            if type_of_chip == "rocm":
+                self.file_suffix = f"{type_of_chip}_{rocm_chip}"
+                self.compile_flags += [
+                    "--iree-hal-target-backends=rocm",
+                    f"--iree-hip-target={rocm_chip}"
+                ]
+                self.benchmark_additional_flags += ["--device_allocator=caching"]
+                self.device = "hip"
+                
+            elif type_of_chip == "cpu":
+                self.file_suffix = "cpu"
```

Comment:
Another code smell from the overspecialization on rocm - "chip" means something very specific for rocm. Other backends don't have that concept or use other words.

Instead prefer to have each test configuration bring its own lists of flags and supporting files, as in, for example https://github.com/iree-org/iree-test-suites/tree/main/onnx_models/configs. Then this code could instead read the lists from the configuration and forward them through to the tests/benchmarks - no logic needed in the test framework itself.